connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -245,34 +245,19 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig)
             ConfigDef basicConfigDef = (connector instanceof SourceConnector)
                                        ? SourceConnectorConfig.configDef()
                                        : SinkConnectorConfig.configDef();
-            Map<String, ConfigValue> validatedConnectorConfig = validateBasicConnectorConfig(
-                    connector,
-                    basicConfigDef,
-                    connectorConfig
-            );
-            configValues.addAll(validatedConnectorConfig.values());
-            configKeys.putAll(basicConfigDef.configKeys());
-            allGroups.addAll(basicConfigDef.groups());
-
-            ConnectorConfig connectorConfigToEnrich = (connector instanceof SourceConnector)
-                    ? new SourceConnectorConfig(plugins(), connectorConfig)
-                    : new SinkConnectorConfig(plugins(), connectorConfig);
-            final ConfigDef connectorConfigDef = connectorConfigToEnrich.enrich(
+
+            ConfigDef newDef = new ConfigDef(basicConfigDef);
+            newDef.embed("", "", configValues.size(), connector.config());
+            ConfigDef connectorConfigDef = ConnectorConfig.enrich(
                     plugins(),
                     basicConfigDef,
                     connectorConfig,
                     false
             );
 
-            // Override is required here after the enriched ConfigDef has been created successfully
+            Config config = new Config(connectorConfigDef.validate(connectorConfig));
             configKeys.putAll(connectorConfigDef.configKeys());
             allGroups.addAll(connectorConfigDef.groups());
-
-            // do custom connector-specific validation
-            Config config = connector.validate(connectorConfig);
```

Comment:
I don't think this is safe to remove since it allows the connector to perform validation beyond what you get from the `ConfigDef`. This method is used both for the validation endpoint and for `putConnectorConfig` where I think you want any exceptions thrown by the connector's validation.